### DISCUSSION TITLE
Yazi is failing to extract .rar files

### DISCUSSION THREAD

**Discussion by poperigby** [sxyazi/yazi#1602] [1. Q&A]
Every time I attempt to extract a .rar file in Yazi, it gets stuck at 99%, and the process reports this error:

```
Micro plugin failed:
runtime error: [string "extract"]:1: 7zip exited when extracting '/home/cassidy/Downloads/StupidMetalHitmarker-55075-1-66-b-1723044651.rar', error code 2
stack traceback:
        [C]: in function 'error'
        [string "extract"]:1: in upvalue 'fail'
        [string "extract"]:66: in method 'try_with'
        [string "extract"]:21: in function <[string "extract"]:14>
```

Version: 0.3.2 (Nixpkgs 2024-08-28)

**Comment by sxyazi** [comment]
Are you sure your 7zip supports extracting `.rar` files? Some distributions/package managers disable RAR support when compiling.

**Reply by poperigby** [reply]
I'm using NixOS with home-manager. I checked the package definition but didn't see any mention of 7z, so I'm not sure.
